gnome-shell-extension-pop-shell: init at 1.0.0

Synthetica9 opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://hydra.nixos.org/job/nixpkgs/trunk/manual/latest/download/1/nixpkgs/manual.html#chap-reviewing-contributions
-->

###### Motivation for this change


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

Synthetica9 committed:
gnome-shell-extension-pop-shell: init at 1.2.0

SuperSandro2000 commented on pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:
```diff
@@ -0,0 +1,32 @@
+  '';
+
+  meta = with stdenv.lib; {
+    description = ''
```
That is the long description. A short one is also needed.

Synthetica9 replied:
Added

luc65r commented on pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:
```diff
@@ -0,0 +1,33 @@
+{ stdenv, lib, fetchFromGitHub, glib, nodePackages }:
```
You don't need `lib` here.

worldofpeace reviewed (changes requested):
Commit message should have `init at` not @

Synthetica9 commented:
> Commit message should have `init at` not @

The commit message already had that, it was only the pr title that didn't. Changed it

SuperSandro2000 reviewed (changes requested):
Can you please squash the commits together?

SuperSandro2000 commented on pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:
```diff
@@ -0,0 +1,33 @@
+      sensible navigation and management of windows. The core feature of Pop
+      Shell is the addition of advanced tiling window management.
+    '';
+    license = licenses.gpl3;
```
homepage is missing here.

SuperSandro2000 commented on pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:
```diff
@@ -0,0 +1,33 @@
+    rmdir $out/usr
+  '';
+
+  meta = with stdenv.lib; {
```
```suggestion
  meta = with lib; {
```

SuperSandro2000 reviewed (changes requested):
please fix the eval error.

Synthetica9 commented:
@grahamc Wait, there was already a PR. Anyways; these are the extensions they use. I updated them to the latest version but can't test that atm since I don't have a Gnome setup (same reason I removed myself as maintainer).

benneti commented on pkgs/desktops/gnome-3/extensions/pop-shell/default.nix:
```diff
@@ -0,0 +1,33 @@
+
+  postInstall = ''
+    mv $out/usr/* $out
+    rmdir $out/usr
```
if you add  ` makeFlags = [ "XDG_DATA_HOME=$(out)/share" ];` you don't need to move the files

benneti reviewed (commented):
Isn't a patch like 
https://github.com/NixOS/nixpkgs/blob/b4f8e0120472735bd0d056f1e1c9c2a14408bdff/pkgs/desktops/gnome-3/extensions/pop-os-shell/fix-gjs.patch
necessary?

genofire commented:
Solved in  #147542 - i believe we could close this PR now.
Maybe you add yourself to maintainers ;)

Synthetica9 commented:
> Solved in #147542 - i believe we could close this PR now. Maybe you add yourself to maintainers ;)

Great to hear, but I don't currently use Gnome. Thanks for picking this up though!
